src/index.js

```diff
@@ -1,10 +1,167 @@
 import { months } from './data.js';
 
 (async function () {
-  const allCityDetails = await fetchCityDetails()
+  let allCityDetails = await fetchCityDetails()
+  const icons = document.querySelectorAll('.mid-ic')
+  const leftArrow = document.getElementById('left-scroll-arrow')
+  const rightArrow = document.getElementById('right-scroll-arrow')
+  let middleSegmentCards = allCityDetails
+  let selectedIcon = ' '
+  let cardCount = document.getElementById('select-icon').value
+  allCityDetails = sortCityOptions(allCityDetails)
   loadSelectOptions(allCityDetails)
   changeHeaderValues()
   document.querySelector('.option-select.city').addEventListener('change', changeHeaderValues)
+  changeMiddleSegment(allCityDetails)
+  getScrollWidth()
+  icons.forEach((ic) => ic.addEventListener('click', selectIcon))
+  leftArrow.addEventListener('click', () => { scrollItems('left') })
+  rightArrow.addEventListener('click', () => { scrollItems('right') })
+  document.getElementById('select-icon').addEventListener('change', getCardCount)
+
+  /**
+   *  This function will get the scrollwidth and clientwidth of the itemscontainer to display the arrow
+   */
+  function getScrollWidth () {
+    const middleContainerWidth = document.getElementsByClassName('items-container')[0].clientWidth
+    const scrollContainerWidth = document.getElementsByClassName('items-container')[0].scrollWidth
```

Comment:
Let's not query two times for the same class, instead have it stored in a variable.
Even within the function scope is fine.